src/Query/Filter/TextField.ts

```diff
@@ -8,23 +8,42 @@ import { FilterOrErrorMessage } from './Filter';
  * value, such as the description or file path.
  */
 export abstract class TextField extends Field {
+    private maybeNegate(match: boolean, filterMethod: String) {
+        return filterMethod.match(/not/) ? !match : match;
+    }
     public createFilterOrErrorMessage(line: string): FilterOrErrorMessage {
         const result = new FilterOrErrorMessage();
         const match = Field.getMatch(this.filterRegexp(), line);
         if (match !== null) {
             const filterMethod = match[1];
-            if (filterMethod === 'includes') {
-                result.filter = (task: Task) =>
-                    TextField.stringIncludesCaseInsensitive(
-                        this.value(task),
-                        match[2],
-                    );
-            } else if (match[1] === 'does not include') {
+            if (['includes', 'does not include'].includes(filterMethod)) {
                 result.filter = (task: Task) =>
-                    !TextField.stringIncludesCaseInsensitive(
-                        this.value(task),
-                        match[2],
+                    this.maybeNegate(
+                        TextField.stringIncludesCaseInsensitive(
+                            this.value(task),
+                            match[2],
+                        ),
+                        filterMethod,
                     );
+            } else if (
+                ['regex matches', 'regex does not match'].includes(filterMethod)
+            ) {
+                // Courtesy of https://stackoverflow.com/questions/17843691/javascript-regex-to-match-a-regex
+                const regexPattern =
+                    /\/((?![*+?])(?:[^\r\n\[/\\]|\\.|\[(?:[^\r\n\]\\]|\\.)*\])+)\/((?:g(?:im?|mi?)?|i(?:gm?|mg?)?|m(?:gi?|ig?)?)?)/;
```

Comment:
I need to forewarn you that a regex this complex, with the current level of test coverage, is going to need a lot of work and tests, in this project, because of the number of times I have seen people 100% confident in the behaviour of their regexes, only for them to have a hidden problem lurking.

### Explanation of the regex

The URL is helpful, but it does mean that everyone who ever reads the code needs to go to Stack Exchange to read it. And if that user's account was ever withdrawn, we would not even be able to do that.

So at a minimum, please include the explanation from SO in the comments here...

### Warnings

WebStorm is warning about this regex:
![image](https://user-images.githubusercontent.com/4840096/183265922-ebb61aa1-b1ca-49e0-bd6f-678472c7d9af.png)

The warnings are:

![image](https://user-images.githubusercontent.com/4840096/183266488-85473ead-69f5-4981-b3c5-f80bbe2e1e8f.png)

Making these changes silenced the warnings:

```diff
-                     /\/((?![*+?])(?:[^\r\n\[/\\]|\\.|\[(?:[^\r\n\]\\]|\\.)*\])+)\/((?:g(?:im?|mi?)?|i(?:gm?|mg?)?|m(?:gi?|ig?)?)?)/;
+                     /\/((?![*+?])(?:[^\r\n[/\\]|\\.|\[(?:[^\r\n\]\\]|\\.)*])+)\/((?:g(?:im?|mi?)?|i(?:gm?|mg?)?|m(?:gi?|ig?)?)?)/;
```

Edit: Indeed, the CI is failing with the same warning:
https://github.com/obsidian-tasks-group/obsidian-tasks/runs/7707526990?check_suite_focus=true#step:5:8

### Test coverage of the regex

I could get all the current tests to pass by stripping the regex all the way back to:

```js
                const regexPattern = /^\/(.*)\/(i?)$/;
```

Which means that none of the other characters are tested, and so either are not working now or are prone to breakage in future.

They don't all need to be tested for every text field, but as `description` is the most flexible field in terms of allowed content, I suggest TDD it - building up to the original SO regex, one failing test at a time.

Here's an example of how I did that for another regex in a PR recently - starting with the first commit by me:

https://github.com/obsidian-tasks-group/obsidian-tasks/pull/976/commits

The tests in the end result were pretty simple:
https://github.com/obsidian-tasks-group/obsidian-tasks/pull/976/files

Reply:
I have let WebStorm refactor to remove the 2 extra `\` that ESLint said were redundant. I opted for the refactoring options that did not change behaviour.